aws_marketplace/using-open-source-model-packages/pytorch-ic-model/notebook.ipynb

```diff
@@ -0,0 +1,461 @@
+    "In this step, we will download class_id_to_label_mapping from S3 bucket. The mapping files has been downloaded from [TensorFlow](https://storage.googleapis.com/download.tensorflow.org/data/ImageNetLabels.txt). [Apache 2.0 License](https://www.apache.org/licenses/LICENSE-2.0)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s3_bucket = f\"jumpstart-cache-prod-{region}\"\n",
+    "key_prefix = \"inference-notebook-assets\"\n",
+    "\n",
+    "def download_from_s3(key_filenames):\n",
+    "    for key_filename in key_filenames:\n",
+    "        s3.download_file(s3_bucket, f\"{key_prefix}/{key_filename}\", key_filename)\n",
+    "\n",
+    "cat_jpg, dog_jpg, ImageNetLabels = \"cat.jpg\", \"dog.jpg\", \"ImageNetLabels.txt\"\n",
+    "\n",
+    "#Download images and label-mapping file.\n",
+    "download_from_s3(key_filenames=[cat_jpg, dog_jpg, ImageNetLabels])\n",
+    "\n",
+    "HTML('<table><tr><td> <img src=\"cat.jpg\" alt=\"cat\" style=\"height: 250px;\"/> <figcaption>cat.jpg</figcaption>'\n",
+    "     '</td><td> <img src=\"dog.jpg\" alt=\"dog\" style=\"height: 250px;\"/> <figcaption>dog.jpg</figcaption>'\n",
+    "     '</td></tr></table>')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!head ImageNetLabels.txt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, open the downloaded images and load them in a variable. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open(ImageNetLabels, 'r') as file: class_id_to_label = file.read().splitlines()[1::] \n",
+    "#The label file has 1001 class labels starting with 'background' class \n",
+    "#but the model predicts 1000 classes sans the background class. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### C. Query endpoint that you have created with the opened images"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#perform_inference method performs inference on the endpoint and prints predictions.\n",
+    "def perform_inference(filename):\n",
+    "    with open(filename, 'rb') as file:\n",
+    "        body=file.read()\n",
+    "        response = runtime.invoke_endpoint(EndpointName=model_name, ContentType=content_type, Body=body)\n",
+    "        prediction= json.loads(response['Body'].read())\n",
+    "        top5_prediction_ids = sorted(range(len(prediction)), key=lambda index: prediction[index], reverse=True)[:5]\n",
+    "        top5_class_labels = \", \".join([class_id_to_label[id] for id in top5_prediction_ids])\n",
+    "        display(HTML(f'<img src={filename} alt={filename} align=\"left\" style=\"width: 250px;vertical-align:middle\"/>' \n",
```

Comment:
Use a different approach than the HTML module.

Reply:
This is what Jumpstart notebook uses. I follow it using same code.
If it is not recommend, can you point me another way?